src/app/(algorithms)/sorting-algorithms/stalin-sort/page.tsx

```diff
@@ -1,3 +1,52 @@
+"use client";
+import GraphLayout from "@/components/GraphLayout/GraphLayout";
+import { createRandomArray, shuffle } from "@/utils/arrays-utils";
+import { useState } from "react";
+
+const initialArray = createRandomArray(15, 10, 100);
+const stalinKills: number[] = [];
+
 export default function Page() {
-  return <div>Stalin</div>;
+  const [isAbleToSort, setIsAbleToSort] = useState(true);
+  const [array, setArray] = useState<number[]>(initialArray);
+  const [chosenColumn, setChosenColumn] = useState(-10);
+
+  const stalinSort = async (array: number[]) => {
+    let comradsArray = [array[0]];
+    setIsAbleToSort(false);
+    for (let i = 1; i < array.length; i++) {
+      if (array[i] >= comradsArray[comradsArray.length - 1]) {
+        comradsArray.push(array[i]);
+      } else {
+        stalinKills.push(array[i]);
+      }
+      setChosenColumn(i);
+      await new Promise((resolve) => setTimeout(resolve, 30000 / sortingSpeed));
+    }
+
+    setIsAbleToSort(true);
+    setArray(comradsArray);
+    stalinKills.length = 0;
+
+    return comradsArray;
+  };
+
+  const [sortingSpeed, setSortingSpeed] = useState<number>(50);
```

Comment:
Move to the top at use those const from above to have same speed convention between every algorithm